lms/djangoapps/program_enrollments/link_program_enrollments.py

```diff
@@ -0,0 +1,184 @@
+""" Management command to link program enrollments and external student_keys to an LMS user """
+import logging
+
+from django.contrib.auth import get_user_model
+from django.db import IntegrityError, transaction
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+from student.models import CourseEnrollmentException
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+NO_PROGRAM_ENROLLMENT_TPL = (u'No program enrollment found for program uuid={program_uuid} and external student '
```

Comment:
While you're making changes here, can you wrap lines over ~100 chars long? After that length they get tough to read on a laptop display with a terminal on the side.

FYI the edX preferred line-wrap style is:
```python
variable = (
    stuff
    more-stuff
    even-more
)